ckanfunctionaltests/api/test_packages.py

```diff
@@ -134,17 +143,21 @@ def test_package_search_by_full_slug_general_term(
             # window)
 
 
+# revision_id is unstable, might have to skip this test to be able to run all the tests?
+# or setup tests which are for dev stacks and staging environments
+@pytest.mark.skip("revision_ids are unstable")
 def test_package_search_by_revision_id_specific_field(
     subtests,
     inc_sync_sensitive,
     base_url_3,
     rsession,
-    random_pkg,
+    stable_pkg,
 ):
     response = rsession.get(
-        f"{base_url_3}/action/package_search?fq=revision_id:{random_pkg['revision_id']}"
+        f"{base_url_3}/action/package_search?fq=revision_id:{stable_pkg['revision_id']}"
```

Comment:
Why are we _replacing_ the randomized tests with the stable ones? I designed it intentionally to allow the randomized tests to augment the "stable" ones side by side so that we would get increased coverage through diversity.

Reply:
I think in situations where the `random_pkg` have been replaced by `stable_pkg` is because it was picking up harvest source and then failing. Running the tests against staging you would have less chance of hitting a harvest sourcee as the datasets will vastly outnumber them, but on my local dev stack where I have 1 harvest source and 2 datasets, there is a 1 in 3 chance of picking up the harvest source package and then failing the test.

For now I'd like the focus be on getting the tests passing on the dev stack reliably, then once we have the mock harvest source available on staging, make sure that the tests are passing there too.

Reply:
But running these tests against stable data doesn't test anything more than the existing stable-data tests. If you just want to run tests against stable-data, just run the stable-data tests, and the entire test suite could have been done in about a quarter the number of LoC. I guess I should have put another config variable in config.json to allow you to do that.

Reply:
The main aim of these tests should be to ensure that we are testing the stability of the response of CKAN between different versions itself not to detect if there are invalid datasets. Or if the test is replicated we could potentially turn off the random tests but most of them are passing. In this case it doesn't pass because if it picks up a harvest source package then it won't match the standard dataset package.